src/algebra/group/hom.lean

```diff
@@ -481,6 +481,15 @@ def flip {mM : monoid M} {mN : monoid N} {mP : comm_monoid P} (f : M →* N →*
   f.flip y x = f x y :=
 rfl
 
+/-- Evaluation of a `monoid_hom` at a point as a monoid homomorphism. See also `monoid_hom.apply`
+for the evaluation of any function at a point. -/
+@[to_additive "Evaluation of an `add_monoid_hom` at a point as an additive monoid homomorphism.
+See also `monoid_hom.apply` for the evaluation of any function at a point."]
+def eval [monoid M] [comm_monoid N] (x : M) : (M →* N) →* N := ⟨λ f, f x, rfl, λ f g, rfl⟩
```

Comment:
Can you go one step further and make this `eval : M →* (M →* N) →* N`?

Reply:
> Or perhaps `(M →* N) →* M →* N` and (...)

And call it `monoid_hom.id`.

Reply:
This right-associativity thing continues to screw with my pattern matching...